File: board-admin-backend/src/main/java/com/portSrilanka/board_admin_backend/dto/paper/PaperRequest.java
package com.portSrilanka.board_admin_backend.dto.paper;

import com.portSrilanka.board_admin_backend.enums.PaperType;
import lombok.Data;

@Data
public class PaperRequest {
    private Long meetingId;
    private Long agendaItemId;
    private PaperType paperType;
    private String title;
    private String referenceNumber;
    private String filePath;
    private String fileName;
    private Integer versionNumber;
    private boolean requiresApproval;
    private boolean isMainPaper;
    private String disclaimerMessage;
}

File: board-admin-frontend/frontend/lib/features/papers/model/paper_request.dart
class PaperRequest {
  final int meetingId;
  final int agendaItemId;
  final String paperType;
  final String title;
  final String? referenceNumber;
  final String? filePath;
  final String? fileName;
  final int? versionNumber;
  final bool requiresApproval;
  final bool isMainPaper;
  final String? disclaimerMessage;

  PaperRequest({
    required this.meetingId,
    required this.agendaItemId,
    required this.paperType,
    required this.title,
    this.referenceNumber,
    this.filePath,
    this.fileName,
    this.versionNumber,
    required this.requiresApproval,
    required this.isMainPaper,
    this.disclaimerMessage,
  });

  Map<String, dynamic> toJson() {
    return {
      'meetingId': meetingId,
      'agendaItemId': agendaItemId,
      'paperType': paperType,
      'title': title,
      'referenceNumber': referenceNumber,
      'filePath': filePath,
      'fileName': fileName,
      'versionNumber': versionNumber,
      'requiresApproval': requiresApproval,
      'isMainPaper': isMainPaper,
      'disclaimerMessage': disclaimerMessage,
    };
  }
}
